go/protocol/keybase1/extras.go

```diff
@@ -2324,25 +2330,48 @@ func (r TeamRole) IsReaderOrAbove() bool {
 	return r.IsOrAbove(TeamRole_READER)
 }
 
+func (r TeamRole) IsBotOrAbove() bool {
+	return r.IsOrAbove(TeamRole_BOT)
+}
+
 func (r TeamRole) IsRestrictedBotOrAbove() bool {
 	return r.IsOrAbove(TeamRole_RESTRICTEDBOT)
 }
 
+func (r TeamRole) IsBotLike() bool {
+	switch r {
+	case TeamRole_BOT, TeamRole_RESTRICTEDBOT:
+		return true
+	}
+	return false
+}
+
 func (r TeamRole) IsRestrictedBot() bool {
 	return r == TeamRole_RESTRICTEDBOT
 }
 
-func (r TeamRole) IsOrAbove(min TeamRole) bool {
+func (r TeamRole) teamRoleForOrderingOnly() int {
 	switch r {
 	case TeamRole_NONE:
-		return min == TeamRole_NONE
+		return 0
 	case TeamRole_RESTRICTEDBOT:
-		return min == TeamRole_NONE || min == TeamRole_RESTRICTEDBOT
+		return 1
+	case TeamRole_BOT:
+		return 2
+	case TeamRole_READER,
+		TeamRole_WRITER,
+		TeamRole_ADMIN,
+		TeamRole_OWNER:
+		return int(r) + 2
 	default:
-		return int(r) >= int(min) || min == TeamRole_RESTRICTEDBOT
+		panic("unhandled teamRole")
```

Comment:
woah there how about 0?

Reply:
oh i thought it would be better to crash since this is a sensitive check

Reply:
Previously it didn't panic and just returned the wrong result. An error return path would be one way to go. I think the panic would do more harm than good imagining some ancillary system that using the comparison on untrusted input. Like some chat doohickey where someone sends a system message "Josh joined the team as a role 10".

Reply:
Heh yeah chat already uses this in a few places, good call. Will switch back to 0